lib/common_test/src/ct_ssh.erl

```diff
@@ -494,438 +582,502 @@ subsystem(SSH, ChannelId, Subsystem, Timeout) ->
 %%%------------------------ SFTP COMMANDS --------------------------
 
 %%%-----------------------------------------------------------------
-%%% @spec sftp_connect(SSH) -> {ok,Server} | {error,Reason}
-%%%      SSH = connection()
-%%%      Server = pid()
-%%%      Reason = term()
 %%% @doc Starts an SFTP session on an already existing SSH connection.
 %%%      <code>Server</code> identifies the new session and must be
 %%%      specified whenever SFTP requests are to be sent.
+%%% @end
+-spec sftp_connect(SSH) -> {ok,Server} | {error,Reason} when
+      SSH :: connection(),
+      Server :: pid(),
+      Reason :: term().
 sftp_connect(SSH) ->
     call(SSH, sftp_connect).
 
 %%%-----------------------------------------------------------------
-%%% @spec read_file(SSH, File) -> Result
-%%%      SSH = connection()
-%%%      Result = ssh_sftp_return() | {error,Reason}
-%%%      Reason = term()
 %%% @doc For info and other types, see ssh_sftp(3).
+-spec read_file(SSH, File) -> ssh_sftp_return() | {error,Reason} when
+      SSH :: connection(),
+      File :: string(),
+      Reason :: term().
 read_file(SSH, File) ->
     call(SSH, {read_file,sftp,File}).
 %%%-----------------------------------------------------------------
-%%% @spec read_file(SSH, Server, File) -> Result
-%%%      SSH = connection()
-%%%      Result = ssh_sftp_return() | {error,Reason}
-%%%      Reason = term()
 %%% @doc For info and other types, see ssh_sftp(3).
+-spec read_file(SSH, Server, File) -> ssh_sftp_return() | {error,Reason} when
+      SSH :: connection(),
+      Server :: atom(),
```

Comment:
Hm... On line 27 it says "The <code>Server</code> argument in the SFTP functions should only be used for SFTP sessions that have been started on existing SSH connections (i.e. when the original connection type is <code>ssh</code>). Whenever the connection type is <code>sftp</code>, use the SSH connection reference only."

I am not saying this is very obvious, but my understanding is that the value <code>sftp</code> is internal. It will be used if you don't specify the <code>Server</code> argument, and according to the above that should be the case if the original connection type is <code>sftp</code>. In that case <code>ct_ssh</code> will pick the <code>SSHRef</code> from its state instead (which is the second element in result of ssh_sftp:start_channel - a pid). Any thoughts?